internal/log/log.go

```diff
@@ -24,9 +25,14 @@ func init() {
 }
 
 // ReplaceGlobalLogger replace the logger and inject it globally
-func ReplaceGlobalLogger(enableJsonOutput bool, verbosityLevel string) error {
+func ReplaceGlobalLogger(enableJsonOutput bool, verbosityLevel string, disableColorEncoder bool) error {
```

Comment:
I think it would be a bit clearer to say `enableColorEncoder` rather than `disableColorEncoder`. This prevents the "double negative" (`!disableColorEncoder`) in the conditional on line 32.

Reply:
I understand not wanting to have the "double negative". But my concern is that changing it to `enableColorEncoder` will change the meaning.

`!enableColorEncoder` will be made to mean `DO NOT enableColorEncoder` when what it's actually doing is `DO enableColorEncoder`.

Reply:
Correct, if `disableColorEncoder` is renamed to `enableColorEncoder`, then the conditional needs to be changed from `if !disableColorEncoder { ...` to `if enableColorEncoder { ...`. Does that make sense? Please let me know if I'm missing something.

Reply:
We only want to `enableColorEncoder` if `cfg.NoColor == false`.  Since the function input is `cfg.NoColor`, then we can't have `if enableColorEncoder { ...`.

It still needs to be negated to `if !enableColorEncoder { ...`

Can we just opt for a neutral name? E.g. `colorEncoder`. I don't think this will be subject to "double negative" or misinterpretation.

Reply:
I am also a bit confused. Why can't the input be passed in as `!cfg.NoColor`?

Reply:
Thought of that. But won't it also be "double negatives"?

Reply:
I suppose. But since all the other flags are written in the affirmative, it seems to me that we can make an exception here to maintain consistency?

I mean, most UNIX commands i.e. `ls`, require the user to explictly state that they want coloured output and if we went that route, we wouldn't have this issue but I am more in favour of having in on by default and so if it means we have the double negative where we pass in the variable, that seems fine to me.

Reply:
True that. I agree.

Any objections @joshklop?

Reply:
Just to be clear, I am in favour of passing in the variable as `!cfg.NoColor` [here](https://github.com/NethermindEth/juno/blob/c33c4175a033dca4bd55284628e738593be96229/cmd/juno/main.go#L233). So that the signature can read as

```go
func ReplaceGlobalLogger(enableJsonOutput bool, verbosityLevel string, enableColorEncoder bool) error
```

and then we can avoid the double negative inside the function.

Reply:
Thank you for clarifying @tshakalekholoane. That is what I was getting at in my original review, but didn't communicate it very clearly.

Reply:
Okay. Fixed it here https://github.com/NethermindEth/juno/pull/335/commits/045d368edf52218c65e681867ffbf1b737367f02

Thanks, guys.